service/resource/endpoint/resource.go

```diff
@@ -79,6 +97,51 @@ func getAnnotations(pod apiv1.Pod, ipAnnotationName string, serviceAnnotationNam
 	return ipAnnotationValue, serviceAnnotationValue, nil
 }
 
+func isEmptyEndpoint(endpoint apiv1.Endpoints) bool {
+	for _, subset := range endpoint.Subsets {
+		if len(subset.Addresses) > 0 {
+			return false
+		}
+	}
+	return true
+}
+
+func (r *Resource) newK8sEndpoint(endpoint Endpoint) (*apiv1.Endpoints, error) {
```

Comment:
Let's please put the methods above the functions.

Reply:
Fixed, I was not aware that we do that.